media_driver/agnostic/gen11/codec/kernel_free/media_srcs.cmake

```diff
@@ -18,6 +18,33 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
+if (BUILD_KERNELS)
+
+    function(gen_cm_kernel_from_source name platform)
+
+        set(cm_krn_dir ${CMAKE_SOURCE_DIR}/media_driver/agnostic/${platform}/codec/kernel_free)
+        set(cm_krn ig${name}krn_g11)
```

Comment:
Use instead:
```
platform_to_genx(${platform} genx kind)
set(cm_krn ig${name}krn_g${genx})
```

You may need to adjust `platform_to_genx` to recognize gen11 (it now knows only gen11_icllp).